lms/djangoapps/discussion/static/discussion/js/discussion_router.js

```diff
@@ -0,0 +1,142 @@
+                        root: '/courses/' + this.courseId + '/discussion/forum/'
+                    });
+                },
+
+                stop: function() {
+                    Backbone.history.stop();
+                },
+
+                allThreads: function() {
+                    this.nav.updateSidebar();
+                    return this.nav.goHome();
+                },
+
+                setActiveThread: function() {
+                    if (this.thread) {
+                        return this.nav.setActiveThread(this.thread.get('id'));
+                    } else {
+                        return this.nav.goHome;
+                    }
+                },
+
+                showThread: function(forumName, threadId) {
+                    this.thread = this.discussion.get(threadId);
+                    this.thread.set('unread_comments_count', 0);
+                    this.thread.set('read', true);
+                    this.setActiveThread();
+                    return this.showMain();
+                },
+
+                showMain: function() {
+                    var self = this;
+                    if (this.main) {
+                        this.main.cleanup();
+                        this.main.undelegateEvents();
+                    }
+                    if (!($('.forum-content').is(':visible'))) {
+                        $('.forum-content').fadeIn();
+                    }
+                    if (this.newPostView.$el.is(':visible')) {
+                        this.newPostView.$el.fadeOut();
+                    }
+                    this.main = new DiscussionThreadView({
+                        el: $('.forum-content'),
+                        model: this.thread,
+                        mode: 'tab',
+                        course_settings: this.course_settings
+                    });
+                    this.main.render();
+                    this.main.on('thread:responses:rendered', function() {
+                        return self.nav.updateSidebar();
+                    });
+                    return this.thread.on('thread:thread_type_updated', this.showMain);
+                },
+
+                navigateToThread: function(threadId) {
+                    var thread;
+                    thread = this.discussion.get(threadId);
+                    return this.navigate('' + (thread.get('commentable_id')) + '/threads/' + threadId, {
+                        trigger: true
+                    });
+                },
+
+                navigateToAllThreads: function() {
+                    return this.navigate('', {
+                        trigger: true
+                    });
+                },
+
+                showNewPost: function() {
+                    var self = this;
+                    return $('.forum-content').fadeOut({
+                        duration: 200,
+                        complete: function() {
+                            return self.newPostView.$el.fadeIn(200).focus();
```

Comment:
Dunno, but this is pre-existing code from [here](https://github.com/edx/edx-platform/pull/12771/files#diff-88736be93e96fb6a15df0da78030b316L177)